### PR TITLE
main: display descriptions of all tools

### LINKED ISSUE #13026: perf-simple-query not listed in --list-tools
```console
$ ./build/dev/scylla --list-tools
types - a command-line tool to examine values belonging to scylla types
sstable - a multifunctional command-line tool to examine the content of sstables
```

where is perf-simple-query?

### PR DESCRIPTION
**Pull request by tchaikov**
- main: expose tools as a vector<>
- main: use a struct for representing tool
- main: track tools descriptin in tool struct
- main: add missing descriptions for tools
- main: move get_tools() into main()

Fixes #13026

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -1756,13 +1756,7 @@ static main_func_type lookup_main_func(std::string_view name) {
         {"perf-simple-query", perf::scylla_simple_query_main},
         {"perf-sstable", perf::scylla_sstable_main},
     };
```
Comment (denesb) [inline main.cc:1763]: At this point this can be just moved to `main()`, no reason to keep in a separate method.

**Review by denesb — APPROVED** [review]
One minor comment, LGTM otherwise.

File: main.cc (hunk 1 above)
Comment (tchaikov) [reply main.cc:1763]: sure, will create another commit to move it into `main()`.

**Comment by tchaikov** [comment]
@denesb hi Botond, updated accordingly. could you please take another look?

**Pull request merged**